Sources/SpeziBluetooth/Bluetooth.swift

```diff
@@ -417,30 +547,120 @@ public actor Bluetooth: Module, EnvironmentAccessible, BluetoothScanner, Bluetoo
+    ///     if we don't hear back from the device. Minimum is 1 second.
+    ///   - autoConnect: If enabled, the bluetooth manager will automatically connect to
     ///     the nearby device if only one is found for a given time threshold.
-    public func scanNearbyDevices(autoConnect: Bool = false) {
+    public func scanNearbyDevices(
+        minimumRSSI: Int? = nil,
+        advertisementStaleInterval: TimeInterval? = nil,
+        autoConnect: Bool = false
+    ) {
         bluetoothManager.assumeIsolated { manager in
-            manager.scanNearbyDevices(autoConnect: autoConnect)
+            manager.scanNearbyDevices(
+                discovery: discoveryConfiguration,
+                minimumRSSI: minimumRSSI,
+                advertisementStaleInterval: advertisementStaleInterval,
+                autoConnect: autoConnect
+            )
         }
     }
 
-    /// If scanning, toggle the auto-connect feature.
-    /// - Parameter autoConnect: Flag to turn on or off auto-connect
-    @_documentation(visibility: internal)
-    public func setAutoConnect(_ autoConnect: Bool) {
+    /// Stop scanning for nearby bluetooth devices.
+    public func stopScanning() {
         bluetoothManager.assumeIsolated { manager in
-            manager.setAutoConnect(autoConnect)
+            manager.stopScanning()
         }
     }
+}
+
+
+extension Bluetooth: BluetoothScanner {
+    func scanNearbyDevices(_ state: BluetoothModuleDiscoveryState) {
+        scanNearbyDevices(
+            minimumRSSI: state.minimumRSSI,
+            advertisementStaleInterval: state.advertisementStaleInterval,
+            autoConnect: state.autoConnect
+        )
+    }
+
+    func updateScanningState(_ state: BluetoothModuleDiscoveryState) {
+        let managerState = BluetoothManagerDiscoveryState(
+            configuredDevices: discoveryConfiguration,
+            minimumRSSI: state.minimumRSSI,
+            advertisementStaleInterval: state.advertisementStaleInterval,
+            autoConnect: state.autoConnect
+        )
 
-    /// Stop scanning for nearby bluetooth devices.
-    public func stopScanning() {
         bluetoothManager.assumeIsolated { manager in
-            manager.stopScanning()
+            manager.updateScanningState(managerState)
         }
     }
 }
+
+// MARK: - Device Handling
+
+extension Bluetooth {
+    func prepareDevice<Device: BluetoothDevice>(id uuid: UUID, _ device: Device.Type, peripheral: BluetoothPeripheral) -> Device {
+        let device = device.init()
+        
+        let didInjectAnything = device.inject(peripheral: peripheral, using: self)
+        if didInjectAnything {
+            initializedDevices[uuid] = device.weaklyReference
+        } else {
+            logger.warning(
+                """
+                \(Device.self) is an empty device implementation. \
+                The same peripheral might be instantiated via multiple \(Device.self) instances if not device property wrappers like
```

Comment:
Might be a typo here?
```suggestion
                The same peripheral might be instantiated via multiple \(Device.self) instances if no device property wrappers like
```